### PR TITLE
clean_up: check for stratisd process before D-Bus calls

### LINKED ISSUE #144: In clean_up, if stratisd isn't even running, don't attempt any D-Bus related cleanup actions
Instead, exit with the info that stratisd just isn't running as a RuntimeError.

### PR DESCRIPTION
**Pull request by bgurney-rh**
Closes #144

### CODE REVIEW

File: testlib/infra.py [hunk 1]
```diff
@@ -51,6 +51,10 @@ def check_result(result, format_str, format_str_args):
             return
         error_strings.append(f"{format_str % format_str_args}: {msg}")
 
+    if process_exists("stratisd") is None:
+        terminate_traces(MONITOR_DBUS_SIGNALS)
```
Comment (mulkieran) [inline testlib/infra.py:55]: I realized we can simplify further.

Delete the terminate_traces() calls everywhere they are now and put one unconditionally right at the start of the clean_up method. We're only running it for cleanup, and stopping it only if it wasn't stopped in the teardown method. No need for it to record all our cleanup actions.

Then move the check for process_exists right after the terminate_traces() call, so it is totally separate from any of the code chunks that do D-Bus based cleanups.

And then this is done.

**Review by mulkieran — CHANGES_REQUESTED** [review]

File: testlib/infra.py (hunk 1 above)
Comment (bgurney-rh) [reply testlib/infra.py:55]: Should these be before or after the `exec_command(['udevadm', 'settle'])` call?

Comment (bgurney-rh) [reply testlib/infra.py]: ...I chose after; it seems sensible.  Updated.

File: testlib/infra.py [hunk 2]
```diff
@@ -43,6 +43,11 @@ def clean_up():
 
     error_strings = []
 
+    terminate_traces(MONITOR_DBUS_SIGNALS)
+
+    if process_exists("stratisd") is None:
+        raise RuntimeError("stratisd process is not running")
+
```
Comment (mulkieran) [inline testlib/infra.py:50]: Plz move ```error_strings = []``` to right before the definition of ```check_result()```. After that, we can consider this complete.

**Review by mulkieran — CHANGES_REQUESTED** [review]

**Comment by bgurney-rh** [comment]
Squashed into a single commit.

File: testlib/infra.py (hunk 2 above)
Comment (bgurney-rh) [reply testlib/infra.py:48]: Resolved.

**Review by mulkieran — APPROVED** [review]

**Pull request merged**